lib/storage/storage.js

```diff
@@ -46,15 +50,29 @@ class StorageBase {
             .replace(/=+$/, '');
     }
 
+    static isCleanText(text) {
+        const lowercased = text.toLowerCase();
+        return profanities.some(badWord => lowercased.includes(badWord));
+    }
+
     handler(req, res) {
         let config = null;
+        // Get the desired config, which we'll first check for profanities in its hash
         if (req.body.config) {
-            config = JSON.stringify(req.body.config);
+            config = req.body.config;
         } else if (req.body.sessions) {
-            config = JSON.stringify(req.body);
+            config = req.body;
         }
-
-        const configHash = StorageBase.safe64Encoded(hash(config, FILE_HASH_VERSION));
+        config.cleanings = 0;
+        // Keep rehashing until no more profanities are found
+        let configHash = StorageBase.safe64Encoded(hash(JSON.stringify(config), FILE_HASH_VERSION));
+        while (!StorageBase.isCleanText(configHash.substr(0, CLEAN_NAME_MAX_LENGTH))) {
+            config.cleanings++; // Shake up the hash a bit
+            logger.info(`Profanity found in full hash ${configHash} - Trying again (${config.cleanings})`);
```

Comment:
This will be handy in debugging if we ever need it. This is of course almost impossible to test!

Reply:
This is why mocks are great. Sinon is a great library for that.

Reply:
True that! But if we have an actual real-world case that comes out as a profanity that works too :grinning: